benchmarks/frontend/scripts/sweep_k8s/metrics.py

```
@@ -0,0 +1,117 @@
+        )
+        pod_name = result.stdout.strip()
+        if not pod_name:
+            return None
+
+        # Exec curl inside the pod (curl may not be available; try wget too)
+        result = subprocess.run(
+            ["kubectl", "-n", namespace, "exec", pod_name, "--",
+             "sh", "-c",
+             f"curl -sf http://{endpoint}/metrics 2>/dev/null || "
+             f"wget -qO- http://{endpoint}/metrics 2>/dev/null || "
+             f"python3 -c \"import urllib.request; print(urllib.request.urlopen('http://{endpoint}/metrics').read().decode())\" 2>/dev/null"],
```

Ruff (0.15.7)

[error] Starting a process with a partial executable path

(S607)